sensor/kubernetes/eventpipeline/component/message.go

```diff
@@ -73,13 +77,47 @@ func (e *ResourceEvent) AddDeploymentForReprocessing(ids ...string) *ResourceEve
 	return e
 }
 
-// AddDeploymentReference creates and sets a new deployment reference to this resource event.
-func (e *ResourceEvent) AddDeploymentReference(reference resolver.DeploymentReference, action central.ResourceAction, force bool) {
-	e.DeploymentReferences = append(e.DeploymentReferences, DeploymentReference{
+// DeploymentReferenceOption option function for the DeploymentReference struct
+type DeploymentReferenceOption func(*DeploymentReference)
+
+// DeploymentRefWithParentResourceAction sets the ParentResourceAction
+func DeploymentRefWithParentResourceAction(action central.ResourceAction) DeploymentReferenceOption {
```

Comment:
I think the prefix is unnecessary. It reads a bit odd on the client call. Let's call it just `WithParentResourceAction`. 

If you don't want to make it confusing for readers that we have different `With*` prefixed functions in the same file, I think it makes sense to move all `DeploymentReference` code into a new file in the same package.

Reply:
Done.